### PR TITLE
Adds some documentation to species.dm

### PR DESCRIPTION
**Pull request by trollbreeder**
## About The Pull Request
Adds documentation to all vars and 40-50% procs in species.dm
Also updates some of the code to use defines instead of 0 and 1

## Why It's Good For The Game
DOCUMENTATION = GOODE
HAVING CODE EXPLAINED = GOODE

### CODE REVIEW

**Comment by trollbreeder** [comment]
SHIT HOLD ON I SCUFFED ONE SECOND

**Comment by trollbreeder** [comment]
Panic over

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1,58 +1,110 @@
...
-	var/wings_icon = "Angel" //the icon used for the wings
+	/// Default mutant bodyparts for this species, like horns. Don't forget to set one for every mutant bodypart you allow this species to have.
+	var/list/default_features = list()
+	/// Visible CURRENT bodyparts that are unique to a species. DO NOT USE THIS AS A LIST OF ALL POSSIBLE BODYPARTS AS IT WILL FUCK SHIT UP! Changes to this list for non-species specific bodyparts (ie cat ears and tails) should be assigned at organ level if possible. Layer hiding is handled by [datum/species/handle_mutant_bodyparts()] below.
+	var/list/mutant_bodyparts = list()
+	///Internal organs that are unique to this race, like a tail.
+	var/list/mutant_organs = list()
+	///This affects the race's speed. Positive numbers make it move slower, negative numbers make it move faster.
+	var/speedmod = 0
+	///Overall defense for the race, or less defense, if it's negative.
```
[the remark this answers is not in the document]
Comment (trollbreeder) [reply code/modules/mob/living/carbon/human/species.dm]: it's clear that the damage mods say multiplier for X damage, but for armor and speed this isn't so obvious, i'll change it

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1,58 +1,110 @@
...
+	var/id
+	//This is used if you want to use a different species' limb sprites.
+	var/limbs_id
+	///This is the fluff name. They are displayed on health analyzers and in the character setup menu. Leave them generic for other servers to customize.
+	var/name
+	// Default color. If mutant colors are disabled, this is the color that will be used by that race.
+	var/default_color = "#FFF"
+
+	///Whether or not the race has sexual characteristics (biological genders). At the moment this is only 0 for skeletons and shadows
+	var/sexes = 1
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
	///Whether or not the race has sexual characteristics (biological genders). At the moment this is only FALSE for skeletons and shadows
	var/sexes = TRUE
```

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -1,58 +1,110 @@
...
-	var/exotic_blood = ""	// If your race wants to bleed something other than bog standard blood, change this to reagent id.
-	var/exotic_bloodtype = "" //If your race uses a non standard bloodtype (A+, O-, AB-, etc)
-	var/meat = /obj/item/reagent_containers/food/snacks/meat/slab/human //What the species drops on gibbing
+	///This allows races to have specific hair colors. If null, it uses the H's hair/facial hair colors. If "mutcolor", it uses the H's mutant_color.
+	var/hair_color
+	///The alpha used by the hair. 255 is completely solid, 0 is invisible.
+	var/hair_alpha = 255
+
+	///Does the species use skintones or not? As of now only used by humans.
+	var/use_skintones = 0
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
	var/use_skintones = FALSE
```

File: code/modules/mob/living/carbon/human/species.dm [hunk 4]
```diff
@@ -1,58 +1,110 @@
...
 	var/disliked_food = GROSS
+	///Food type that is toxic to the species, giving them toxin damage. Meat is toxic to moths, for example.
 	var/toxic_food = TOXIC
-	var/list/no_equip = list()	// slots the race can't equip stuff to
-	var/nojumpsuit = 0	// this is sorta... weird. it basically lets you equip stuff that usually needs jumpsuits without one, like belts and pockets and ids
-	var/say_mod = "says"	// affects the speech message
+	/////Inventory slots the race can't equip stuff to. Golems cannot wear jumpsuits, for example.
+	var/list/no_equip = list()
+	/// Allows the species to equip items that normally require a jumpsuit without having one equipped. Used by golems.
+	var/nojumpsuit = 0
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
	var/nojumpsuit = FALSE
```

**Review by nemvar — COMMENTED** [review]
I highlighted some things that are booleans but are not using the defines.
Would be nice if you also updated them to current standards.

File: code/modules/mob/living/carbon/human/species.dm [hunk 5]
```diff
@@ -120,11 +189,26 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
 		return TRUE
 	return FALSE
 
+/**
+  * Generates a random name for a carbon.
+  *
+  * This generates a random unique name based on a human's species and gender.
+  * Arguments:
+  * gender - The gender that the name should adhere to. Use MALE for male names, use anything else for female names.
+  * unique -
```
Comment (trollbreeder) [inline code/modules/mob/living/carbon/human/species.dm]: Note to self: Find out what unique does and add it to this documentation

**Comment by cacogen** [comment]
egood

File: code/modules/mob/living/carbon/human/species.dm [hunk 6]
```diff
@@ -1,58 +1,110 @@
...
+	var/use_skintones = FALSE
+	///If your race bleeds something other than bog standard blood, change this to reagent id. For example, ethereals bleed liquid electricity.
+	var/exotic_blood = ""
+	///If your race uses a non standard bloodtype (A+, O-, AB-, etc). For example, lizards have L type blood.
+	var/exotic_bloodtype = ""
+	///What the species drops when gibbed by a gibber machine.
+	var/meat = /obj/item/reagent_containers/food/snacks/meat/slab/human
+	///What skin the species drops when gibbed by a gibber machine.
 	var/skinned_type
+	///Food type that the species likes, giving them a mood boost. Lizards like meat, for example.
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/human/species.dm]: pluralize type to types else people might not realize its a bitfield, on that note, maybe mention its a bitfield.

File: code/modules/mob/living/carbon/human/species.dm [hunk 7]
```diff
@@ -1,58 +1,110 @@
...
 	var/liked_food = NONE
+	///Food type that the species dislikes, giving them disgust. Humans hate raw food, for example.
 	var/disliked_food = GROSS
+	///Food type that is toxic to the species, giving them toxin damage. Meat is toxic to moths, for example.
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/human/species.dm]: this is not what toxic category food does; it's a shit name (which is my bad) but its generally closed to hated foods, the toxic damage is usually the reagents themselves. It just raises disgust more than disliked food

File: code/modules/mob/living/carbon/human/species.dm [hunk 8]
```diff
@@ -1,58 +1,110 @@
...
 
-	var/hair_color	// this allows races to have specific hair colors... if null, it uses the H's hair/facial hair colors. if "mutcolor", it uses the H's mutant_color
-	var/hair_alpha = 255	// the alpha used by the hair. 255 is completely solid, 0 is transparent.
-
-	var/use_skintones = 0	// does it use skintones or not? (spoiler alert this is only used by humans)
-	var/exotic_blood = ""	// If your race wants to bleed something other than bog standard blood, change this to reagent id.
-	var/exotic_bloodtype = "" //If your race uses a non standard bloodtype (A+, O-, AB-, etc)
-	var/meat = /obj/item/reagent_containers/food/snacks/meat/slab/human //What the species drops on gibbing
+	///This allows races to have specific hair colors. If null, it uses the H's hair/facial hair colors. If "mutcolor", it uses the H's mutant_color.
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/human/species.dm]: I forgot to add this to this comment but "fixedmutcolor" makes it use fixedmutcolor.

File: code/modules/mob/living/carbon/human/species.dm [hunk 9]
```diff
@@ -1,58 +1,110 @@
...
+	///Lowest possible punch damage this species can give. If this is set to 0, punches will always miss.
+	var/punchdamagelow = 1
+	///Highest possible punch damage this species can give.
+	var/punchdamagehigh = 10
+	///Damage at which punches from this race will stun
+	var/punchstunthreshold = 10 //yes it should be to the attacked race but it's not useful that way even if it's logical
+	///Base electrocution coefficient.  Basically a multiplier for damage from electrocutions.
+	var/siemens_coeff = 1
+	///What kind of damage overlays (if any) appear on our species when wounded?
+	var/damage_overlay_type = "human"
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/human/species.dm]: mention that "" means no overlays.

File: code/modules/mob/living/carbon/human/species.dm (hunk 5 above)
Comment (Qustinnus) [reply code/modules/mob/living/carbon/human/species.dm]: It tries to ensure that the name is unique and no dupes exist in the station rn

File: code/modules/mob/living/carbon/human/species.dm [hunk 10]
```diff
@@ -142,11 +226,24 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 
 	return randname
 
-//Called when cloning, copies some vars that should be kept
+/**
+  * Called when cloning, copies some vars that should be kept
+  *
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
/**
```
Cloning is kill

File: code/modules/mob/living/carbon/human/species.dm [hunk 11]
```diff
@@ -120,11 +189,26 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
 	return FALSE
 
+/**
+  * Generates a random name for a carbon.
+  *
+  * This generates a random unique name based on a human's species and gender.
+  * Arguments:
+  * gender - The gender that the name should adhere to. Use MALE for male names, use anything else for female names.
+  * unique - If true, ensures that this new name is not a duplicate of anyone else's name currently on the station.
+  * lastname - Does this species' naming system adhere to the last name system? Set to false if it doesn't.
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/human/species.dm]: Be a dear please and make the arguments use the format specified in the autodoc guide(* * arg1 - why is this relevant)

File: code/modules/mob/living/carbon/human/species.dm (hunk 10 above)
Comment (trollbreeder) [reply code/modules/mob/living/carbon/human/species.dm]: Yeah but this is still used by some species, like slime dudes

File: code/modules/mob/living/carbon/human/species.dm (hunk 11 above)
Comment (trollbreeder) [reply code/modules/mob/living/carbon/human/species.dm]: They're in the order from arg1 to arg3, and i named them after the variables that they're used in the proc. I think arg1 is simply a placeholder for the argument's name.

File: code/modules/mob/living/carbon/human/species.dm (hunk 10 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/human/species.dm]: Yes leave that bit I'm just saying to remove the reference to cloning

File: code/modules/mob/living/carbon/human/species.dm (hunk 11 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/human/species.dm]: Capture
You need two asterisks for arguments, see example

Comment (trollbreeder) [reply code/modules/mob/living/carbon/human/species.dm]: i'll do that, even though regenerate_organs doesn't do this

File: code/modules/mob/living/carbon/human/species.dm (hunk 10 above)
Comment (trollbreeder) [reply code/modules/mob/living/carbon/human/species.dm]: cloning is what i'd best describe what this proc does even if cloning is kill

Comment (TiviPlus) [reply code/modules/mob/living/carbon/human/species.dm]: "Called when cloning" describes when the proc is called and not what it does, so at least remove that bit, though "copies some vars that should be kept" can be left in

Comment (trollbreeder) [reply code/modules/mob/living/carbon/human/species.dm]: i guess that makes sense, tivi

File: code/modules/mob/living/carbon/human/species.dm [hunk 12]
```diff
@@ -1,58 +1,110 @@
...
+	///Bitfield for food types that the species likes, giving them a mood boost. Lizards like meat, for example.
 	var/liked_food = NONE
+	///Bitfield for food types that the species dislikes, giving them disgust. Humans hate raw food, for example.
 	var/disliked_food = GROSS
+	///Bitfield for food types that the species absolutely hates, giving them even more disgust than disliked food. Meat is "toxic" to moths, for example.
 	var/toxic_food = TOXIC
-	var/list/no_equip = list()	// slots the race can't equip stuff to
-	var/nojumpsuit = 0	// this is sorta... weird. it basically lets you equip stuff that usually needs jumpsuits without one, like belts and pockets and ids
-	var/say_mod = "says"	// affects the speech message
+	/////Inventory slots the race can't equip stuff to. Golems cannot wear jumpsuits, for example.
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
	///Inventory slots the race can't equip stuff to. Golems cannot wear jumpsuits, for example.
```

**Review by Rohesie — DISMISSED** [review]

**Pull request merged**